### PR TITLE
sci-libs/xblas: EAPI upgraded to 7

### PR DESCRIPTION
**Pull request by henrikp**
Removed doc IUSE due to having no real effect on the emerge.
Slight corrections to the original suggested package.

Closes: https://bugs.gentoo.org/699382
Package-Manager: Portage-3.0.1, Repoman-2.3.23
RepoMan-Options: --force
Signed-off-by: Henrik Pihl <ahvenas@gmail.com>

### CODE REVIEW

**Comment by TheChymera** [comment]
@henrikp the build fails over here with https://ppb.chymera.eu/d4dfbf.log Some missing file...

**Comment by henrikp** [comment]
I don't have access to the log file?

Kontakt Horea Christian (<notifications@github.com>) kirjutas kuupäeval T,
20. oktoober 2020 kell 23:34:

**Comment by TheChymera** [comment]
@henrikp apologies, It's my self-hoste pastebin software and it keeps doing that when pasting root-owned files... Fixed now, both the link and the software ^^

File: sci-libs/xblas/xblas-1.0.248-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+static_to_shared() {
+	local libstatic=${1}; shift
+	local libname=$(basename ${libstatic%.a})
+	local soname=${libname}$(get_libname $(ver_cut 1-2))
+	local libdir=$(dirname ${libstatic})
+
+	einfo "Making ${soname} from ${libstatic}"
+	if [[ ${CHOST} == *-darwin* ]] ; then
+		${LINK:-$(tc-getCC)} ${LDFLAGS}  \
```
Comment (epsilon-0) [inline sci-libs/xblas/xblas-1.0.248-r2.ebuild:33]: Replace `${LINK:-$(tc-getCC)}` by `$(tc-getLD)`, we want to use the linker configured by the user, which is present in `$(tc-getLD)`.

File: sci-libs/xblas/xblas-1.0.248-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		-dynamiclib -install_name "${EPREFIX}"/usr/lib/"${soname}" \
+		-Wl,-all_load -Wl,${libstatic} \
+		"$@" -o ${libdir}/${soname} || die "${soname} failed"
+	else
+		local components=($(ver_rs 1- ' '))
+
+		${LINK:-$(tc-getCC)} ${LDFLAGS}  \
```
Comment (epsilon-0) [inline sci-libs/xblas/xblas-1.0.248-r2.ebuild:40]: Same here, replace `LINK` by the linker from toolchain.

File: sci-libs/xblas/xblas-1.0.248-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		-shared -Wl,-soname=${soname} \
+		-Wl,--whole-archive ${libstatic} -Wl,--no-whole-archive \
```
Comment (epsilon-0) [inline sci-libs/xblas/xblas-1.0.248-r2.ebuild:42]: whole archive, followed by no whole archive...
seems kinda sus, but I'm not a linker expert.

**Review by epsilon-0 — CHANGES_REQUESTED** [review]
This is a really old ebuild that needs to be updated to proper standards, so I've left some comments for helping fix things.

Thanks a lot for helping update the repository :smile_cat:

**Comment by TheChymera** [comment]
@henrikp would you be interested in incorporating the listed suggestions? We're happy to help if anything is unclear about them or you require further assistance.

**Comment by henrikp** [comment]
Yes, but will get on to that (and some other things) of in half a month or
so due to lack of time currently.

Kontakt Horea Christian (<notifications@github.com>) kirjutas kuupäeval R,
27. november 2020 kell 20:18:

**Comment by TheChymera** [comment]
@henrikp bump? :3

**Comment by AndrewAmmerlaan** [comment]
I think we can close this right? Since xblas has been bumped to EAPI 7 in some other commit, or are their any other import changes in this patch?

**Pull request closed without merging**